File: src/my_robot_controller/scripts/pose_subscriber.py
#!/usr/bin/env python3

import rospy
from turtlesim.msg import Pose


def pose_callback(msg: Pose):
    rospy.loginfo(f"({msg.x:.3f}, {msg.y:.3f})")



def main():
    rospy.init_node("pose_subscriber")
    sub = rospy.Subscriber("/turtle1/pose", Pose, callback=pose_callback)
    
    rospy.loginfo("Node has been started")

    rospy.spin()


if __name__ == "__main__":
    main()

File: src/rosrust_controller/src/pose_subscriber.rs
use rosrust_msg::turtlesim;

const NODE_NAME: &str = "rosrust_pose_subscriber";

fn pose_callback(pose: &turtlesim::Pose) {
    let log = format!("({}, {})", pose.x, pose.y);
    rosrust::ros_info!("{log}");
}

fn main() {
    env_logger::init();
    rosrust::init(NODE_NAME);

    let _subscriber = rosrust::subscribe("/turtle1/pose", 10, move |pose: turtlesim::Pose| {
        pose_callback(&pose);
    })
    .unwrap();

    rosrust::ros_info!("Node has been started");

    rosrust::spin();
}
